### ISSUE TITLE
Migration disable-angular-eslint-prefer-standalone fails

### ISSUE THREAD

**Issue by phihochzwei** [nrwl/nx#29251]
### Current Behavior

When trying to migrate to 20.2.1 from 20.1.2 I receive the following error 

 _NX   Failed to run disable-angular-eslint-prefer-standalone from @nx/angular. This workspace is NOT up to date!
 NX   Cannot find configuration for 'deich-lib'_

If I remove the the migration, all works fine, but the rule is active and breaks our project. And for some reaon I cannot disable it manually. 

### Expected Behavior

Well, I would expect the migration to run as smoothly as always :)

### Steps to Reproduce

1. nx migrate latest
2. nx migrate --run-migrations

### Nx Report

```shell
Node           : 22.6.0
OS             : darwin-arm64
Native Target  : aarch64-macos
npm            : 10.8.2

nx                 : 20.2.1
@nx/js             : 20.2.1
@nx/jest           : 20.2.1
@nx/eslint         : 20.2.1
@nx/workspace      : 20.2.1
@nx/angular        : 20.2.1
@nx/cypress        : 20.2.1
@nx/devkit         : 20.2.1
@nx/eslint-plugin  : 20.2.1
@nx/nest           : 20.2.1
@nx/node           : 20.2.1
@nx/plugin         : 20.2.1
@nx/storybook      : 20.2.1
@nx/web            : 20.2.1
@nx/webpack        : 20.2.1
typescript         : 5.6.3
---------------------------------------
Registered Plugins:
@nx/storybook/plugin
---------------------------------------
Community plugins:
@ngrx/component      : 18.1.1
@ngrx/effects        : 18.1.1
@ngrx/entity         : 18.1.1
@ngrx/router-store   : 18.1.1
@ngrx/schematics     : 18.1.1
@ngrx/signals        : 18.1.1
@ngrx/store          : 18.1.1
@ngrx/store-devtools : 18.1.1
@storybook/angular   : 8.4.7
ng-mocks             : 14.13.1
ng2-charts           : 6.0.1
---------------------------------------
Local workspace plugins:
         @deich-lib/workspace-plugin
```

### Failure Logs

```shell

```

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by leosvelperez** [comment]
Thanks for reporting this! Unfortunately, the information provided is not enough to troubleshoot this issue. Please provide a repo where the issue can be reproduced so we can troubleshoot it.

**Comment by phihochzwei** [comment]
Creating a repo seems a bit problematic since our project is rather large and "old" :) 

But I think I found the problem, even though I am not experienced enough with that special topic: 

Newer NX repos have a file called **elint.config.js** while our project has only a file called **.eslintrc.json**

That would, at least from my perspective, explain the "NX Cannot find configuration file"-message

**Comment by phihochzwei** [comment]
I just ran nx g @nx/eslint:convert-to-flat-config and got the eslint.config.js, but the error still persists. Any idea on what could cause it ?

**Comment by phihochzwei** [comment]
I just fumbled a bit with the code and I found, that the readProjectConfiguration()-function is called with the base folder of the monorepo (which is 'deichlib') and is looking there for a project.json, which for obvious reasons isn't there and than the error occurs

**Comment by andgdk** [comment]
I encountered a similar issue. Most likely, because I was working in another NX workspace previously (where the update completed successfully before). I think they have not been running in parallel.

However, when I tried to reset the deamon (`nx reset`) an error was thrown hinting that some files were locked by another process.
So I restarted my machine and ran `nx reset` for good measure. After this, the migration run just fine (`nx migrate --run-migrations`). Hope this might hel you as well @phihochzwei.

**Comment by gmamisashvili** [comment]
> I just fumbled a bit with the code and I found, that the readProjectConfiguration()-function is called with the base folder of the monorepo (which is 'deichlib') and is looking there for a project.json, which for obvious reasons isn't there and than the error occurs

Did you manage to fix this issue? I have similar issue with git submodule

**Comment by phihochzwei** [comment]
Not yet, since I try migrating aside daily business. But I try to look into it again today

**Comment by phihochzwei** [comment]
Okay I tried it again :) 

This seems somehow be related to the eslint config, because after I removed "disable-angular-eslint-prefer-standalone" I receive the exact same error for "remove-angular-eslint-rules"

**Comment by phihochzwei** [comment]
Okay, I dont think I am stuck here and I need some advice from the team. Here are my findings: 

- The migration tries to fetch all projects (apps, libs etc.) via 'getProjectsFilteredByDependencies'
- The returned list also includes the mono-repo itself
- In the process, the method 'readProjectConfiguration' is called
- This stores the response of the method  'readAndCombineAllProjectConfigurations' in a var called 'allProjects'
- 'allProjects' does NOT contain the monorepo
- Since there is nothing found with the name of the monorepo inside 'allProjects', the method is checking for an 'angular.json', which obviously is not in the root of the mono-repo
- The migration exits with  throw new Error(`Cannot find configuration for '${projectName}'`)



I know, having a repo which reproduces this would help a lot. Unfortunately I am not able to create a monorepo which has this problem. My best guess is, that something has changed in the configuration or its structure during the last 3 years which causes this (Project was startet in November 2021). 

Has ANYONE ANY Idea ?

**Comment by Destreyf** [comment]
TLDR: You can use the command `nx g @nx/eslint:convert-to-flat-config` to run a generator that converts your `.eslintrc.json` files to `eslint.config.cjs` files and then the migration to `20.2.X` or in my case `20.3.X` should work.

---

I have a similar, but not exactly the same issue.  `@nx/angular:disable-angular-eslint-prefer-standalone` fails, but my error states `The "path" argument must be of type string. Received undefined`.

Digging through the error happens due to undefined being passed to the normalize method when resolving the paths.

```
Running migration @nx/angular: disable-angular-eslint-prefer-standalone

 NX   Failed to run disable-angular-eslint-prefer-standalone from @nx/angular. This workspace is NOT up to date!


 NX   The "path" argument must be of type string. Received undefined

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at join (node:path:1251:7)
    at FsTree.normalize (/Users/destreyf/<path-to-project>/node_modules/nx/src/generators/tree.js:227:64)
    at FsTree.read (/Users/destreyf/<path-to-project>/node_modules/nx/src/generators/tree.js:26:25)
    at lintConfigHasOverride (/Users/destreyf/<path-to-project>/node_modules/@nx/eslint/src/generators/utils/eslint-file.js:227:30)
    at default_1 (/Users/destreyf/<path-to-project>/node_modules/@nx/angular/src/migrations/update-20-2-0/disable-angular-eslint-prefer-standalone.js:17:53)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runNxMigration (/Users/destreyf/<path-to-project>/node_modules/nx/src/command-line/migrate/migrate.js:1037:5)
    at async executeMigrations (/Users/destreyf/<path-to-project>/node_modules/nx/src/command-line/migrate/migrate.js:922:33)
    at async runMigrations (/Users/destreyf/<path-to-project>/node_modules/nx/src/command-line/migrate/migrate.js:1010:37)
    at async /Users/destreyf/<path-to-project>/node_modules/nx/src/command-line/migrate/migrate.js:1051:13
```

When digging in further, I find that it's checking for the existence of `eslint.config.(c)js`, however the project it's accessing only has a `.eslintrc.json` file.

This lead to look for past migrations for the `.eslintrc.json`, and I ran across a generator `nx g @nx/eslint:convert-to-flat-config` that converts from `.eslintrc.json` files to `eslint.config.cjs` files, after running the generator the migration successfully completes.

Due to the fix being tied to having to run an apparently optional generator I think this is a bug that needs to be addressed within the migrations themselves, either support for `.eslintrc.json` files needs to be added to the migration, or make the generator a required step if it finds `.eslintrc.json` files.

**Comment by moi-j** [comment]
I was facing the same issue and managed to solve it by changing the name value on `package.json` to my main app name.

E.g.: I have a `package.json` with the name value `nx-workspace`, and then an app called `angular-app`. If I change `nx-workspace` to `angular-app` the migrations run smoothly. After that, I've just rename back the `package.json` name property.

I don't know why I need to name the `package.json` with an existing app name, though. Is that absolutely needed?

**Comment by isawk** [comment]
@Destreyf solution worked for me...

npx nx g @nx/eslint:convert-to-flat-config
...

Than run Migration again...
